### PR TITLE
Feat: Redux-Toolkit implementation for user

### PR DESCRIPTION
**Pull request by Kozgor**
- Redux toolkit implementation
- Minor refactoring
- Changed some tests to handle redux flow
- Removed unused dependencies

### CODE REVIEW

File: src/components/Board/Board.tsx [hunk 1]
```diff
@@ -33,28 +34,30 @@ const Board = (props: {
...
-            avatar={column.avatar}
-            cards={column.cards}
-          />
-        ))}
-      </main>
-    </>
+    <div className={classes['board-container']}>
+      <BoardProvider>
+        <BoardHeader
+          fullName={props.fullName}
```
Comment (LinnykS) [inline src/components/Board/Board.tsx:40]: Major: Better to exclude User-related stuff from BordHeader and use Redux for getting needed information directly in the component.

File: src/context/board/board-context.tsx [hunk 2]
```diff
@@ -1,16 +1,16 @@
 import { createContext, useReducer } from 'react';
-import { BaseProps } from '../interfaces/baseProps';
-import ColumnReducer from './column-reducer';
+import { BaseProps } from '../../interfaces/baseProps';
+import ColumnReducer from './board-reducer';
```
Comment (Yura935) [inline src/context/board/board-context.tsx:3]: Minor: Please rename ColumnReducer to BoardReducer

File: src/components/Board/Board.tsx (hunk 1 above)
Comment (Kozgor) [reply src/components/Board/Board.tsx]: Done

File: src/components/BoardHeader/BoardHeader.test.tsx [hunk 3]
```diff
@@ -5,48 +5,78 @@ import {
...
   });
 
-  test('component mounts properly', () => {
+  test('Should mount component properly', () => {
     expect(component).toBeTruthy();
   });
 
-  test('display username', () => {
-    const username = screen.getByText(/Jack/i);
+  test('Should be \'Sign out\' button in the document', () => {
```
Comment (LinnykS) [inline src/components/BoardHeader/BoardHeader.test.tsx:63]: Minor: I think double quotes can be used for Sign out wrapping.

**Review by LinnykS — APPROVED** [review]

File: src/components/BoardHeader/BoardHeader.test.tsx [hunk 4]
```diff
@@ -5,48 +5,78 @@ import {
...
     fireEvent.click(signOutButton);
 
-    expect(onSignMock).toHaveBeenCalled();
+    expect(removeLocalUserData).toHaveBeenCalled();
+    expect(removeUserFromStore).toHaveBeenCalled();
   });
 
-  test('display `Start` button for timer', () => {
-    const startButton = screen.getByText(/Start/i);
+  test('Should display `Start Timer` button for timer', () => {
```
Comment (LinnykS) [inline src/components/BoardHeader/BoardHeader.test.tsx:78]: Minor: Double quets for "Start Timer" here as well. For consistency.

**Pull request merged**